### PR TITLE
Replace Search and Search Results with FSE templates

### PR DESCRIPTION
**Pull request by Gentyspun**
Ref: https://github.com/amnestywebsite/humanity-theme/issues/373, https://github.com/amnestywebsite/humanity-theme/issues/385

Replaces the old search.php and searchpage.php templates with HTML equivalents, and the necessary patterns to contain the logic.

**Steps to test**:
1. Navigate to /search
2. Check everything looks correct, use the filters to filter the posts and make sure they work correctly
3. Clear all filters, and enter a search query
4. The results should display correctly
5. Use the filters again and check they filter the results
6. Navigate to the site editor and check the templates for the search and searchpage
7. They should match the front end

**Considerations**:
- Accessibility
  - Regions (semantic HTML tags, e.g. `<section>`; [landmarks](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#landmark_roles); etc.)
  - Screen reader compatibility, labels ([`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label), [`aria-labelledby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby), etc.)
  - Keyboard navigability
- Localisation
- RTL

Example: http://bigbite.im/v/Qyiwjq

### CODE REVIEW

File: wp-content/themes/humanity-theme/patterns/form-input.php
Comment (jaymcp) [inline wp-content/themes/humanity-theme/patterns/form-input.php:1]: This is named and labelled as a generic form input, but is specifically for search, so I think we should specify that.

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:1]: Is this to limit the terms outputted to the first term returned?
If so, I think we can avoid having to do this by adding a filter in the pattern.
i.e. pattern: https://github.com/amnestywebsite/humanity-theme/pull/379/files#diff-3ee6d5a5226d4879cb255c132d4b54e642e290046af26bdf0a71561e0ccfaf2fR10
callback: https://github.com/amnestywebsite/humanity-theme/pull/379/files#diff-0ca78843e27cc3f93faa5361b948c9b9ffcb2b942ddfa3f43283e9df37e093edR863-R885

File: wp-content/themes/humanity-theme/patterns/taxonomy-filters.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * Description: Taxonomy filters pattern for the theme
+ * Slug: amnesty/taxonomy-filters
+ * Inserter: no
+ */
+
+?>
+
+<!-- wp:group {"tagName":"div","className":"taxonomyArchive-filters"} -->
+<div class="wp-block-group taxonomyArchive-filters">
+<!-- wp:pattern {"slug":"amnesty/custom-select"} /-->
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/patterns/taxonomy-filters.php:14]: I can't locate this pattern; could you point me to it, please?

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+		return $settings;
+	}
+}
+
+if ( ! function_exists( 'amnesty_render_block_core_post_terms' ) ) {
+	/**
+	 * Renders the `core/query-pagination-next` block on the server.
+	 *
+	 * @since 5.8.0
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:28]: This is incorrect

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		}
+	
+		$suffix = '</div>';
+		if ( isset( $attributes['suffix'] ) && $attributes['suffix'] ) {
+			$suffix = '<span class="wp-block-post-terms__suffix">' . $attributes['suffix'] . '</span>' . $suffix;
+		}
+		
+		return (
+			'<div ' . $wrapper_attributes . '>' .
+				'<a href="' . get_term_link( $post_terms[0] ) . '" rel="tag">' . esc_html( $post_terms[0]->name ) . '</a>' .
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:81]: The href needs escaping

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		}
+		if ( isset( $attributes['style']['elements']['link']['color']['text'] ) ) {
+			$classes[] = 'has-link-color';
+		}
+	
+		$separator = '';
+	
+		$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => implode( ' ', $classes ) ) );
+	
+		$prefix = "<div $wrapper_attributes>";
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:69]: This needs escaping (`wp_kses_data`)

**Review by jaymcp — CHANGES_REQUESTED** [review]
Thanks for all the work you've put into this!

There's a few issues with the pagination:
- The search page doesn't actually paginate
- The `next` link is not clickable, regardless of the current page number, on the search page
- The `previous` link isn't greyed out on page 1 on both search and search results
- The `previous` link loses its arrow on pages greater than 1
- The search page URIs are odd - `/search/?query-0-page-1`
- It sometimes breaks onto two lines (see screenshot)




Additionally, the layout is a little off in the site editor:

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php (hunk 3 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:81]: Resolved

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php (hunk 2 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:28]: Resolved

File: wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php (hunk 4 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/core-blocks/post-terms/filters.php:69]: Resolved

File: wp-content/themes/humanity-theme/patterns/taxonomy-filters.php (hunk 1 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/patterns/taxonomy-filters.php:14]: The pattern mentioned and the file it is included in are unused patterns so I have removed them from the PR. I must have missed them before PR'ing

File: wp-content/themes/humanity-theme/patterns/form-input.php
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/patterns/form-input.php:1]: This actually looks to be unused anywhere, so removed it from the PR

**Pull request merged**